crates/backend/src/codegen.rs

```diff
@@ -1024,114 +1024,115 @@ impl ToTokens for ast::ImportType {
+                    match val {
+                        #(#variant_indices => #variant_paths_ref,)*
+                        #invalid => #enum_name::__Invalid,
+                        _ => unreachable!("The JS binding should only ever produce a valid value or the specific 'invalid' value")
+                    }
                 }
             }
 
             #[automatically_derived]
-            impl #wasm_bindgen::convert::FromWasmAbi for #name {
-                type Abi = <#wasm_bindgen::JsValue as #wasm_bindgen::convert::FromWasmAbi>::Abi;
-
-                unsafe fn from_abi(js: Self::Abi) -> Self {
-                    let s = <#wasm_bindgen::JsValue as #wasm_bindgen::convert::FromWasmAbi>::from_abi(js);
-                    #name::from_js_value(&s).unwrap_or(#name::__Nonexhaustive)
-                }
+            impl #wasm_bindgen::convert::OptionFromWasmAbi for #enum_name {
+                #[inline]
+                fn is_none(val: &u32) -> bool { *val == #hole }
             }
 
             #[automatically_derived]
-            impl #wasm_bindgen::convert::OptionIntoWasmAbi for #name {
+            impl #wasm_bindgen::convert::OptionIntoWasmAbi for #enum_name {
                 #[inline]
-                fn none() -> Self::Abi { <::js_sys::Object as #wasm_bindgen::convert::OptionIntoWasmAbi>::none() }
+                fn none() -> Self::Abi { #hole }
             }
 
             #[automatically_derived]
-            impl #wasm_bindgen::convert::OptionFromWasmAbi for #name {
-                #[inline]
-                fn is_none(abi: &Self::Abi) -> bool { <::js_sys::Object as #wasm_bindgen::convert::OptionFromWasmAbi>::is_none(abi) }
+            impl #wasm_bindgen::describe::WasmDescribe for #enum_name {
+                fn describe() {
+                    use #wasm_bindgen::describe::*;
+                    inform(IMPORT_ENUM);
+                    inform(#name_len);
+                    #(inform(#name_chars);)*
+                    inform(#invalid);
+                    inform(#hole);
+                    inform(#variant_count);
+                    #(#describe_variants)*
+                }
             }
 
             #[automatically_derived]
-            impl From<#name> for #wasm_bindgen::JsValue {
-                fn from(obj: #name) -> #wasm_bindgen::JsValue {
-                    #wasm_bindgen::JsValue::from(obj.to_str())
+            impl #wasm_bindgen::__rt::core::convert::From<#enum_name> for
+                #wasm_bindgen::JsValue
+            {
+                fn from(val: #enum_name) -> Self {
+                    #wasm_bindgen::JsValue::from_str(
+                        match val {
+                            #(#variant_paths_ref => #variant_values,)*
+                            #enum_name::__Invalid => #wasm_bindgen::throw_str("Converting an invalid import enum back to a string is currently not supported"),
+                            _ => unreachable!("All possible variants should have been checked")
```

Comment:
Why is this line necessary?

Reply:
it's needed because the enum is [marked as exhaustive](https://github.com/rustwasm/wasm-bindgen/pull/3915/files?diff=split&w=0#diff-5b89f2e6e335fbc4662194fa0fe557251f21ef1c7a18d50db8e63a30851ca3bbR1062)

Reply:
Non-exhaustive enums can be matched exhaustively inside the crate they are defined in.
So this should not be required, unless I'm missing something here.

Reply:
Ah yeah, forgot about that detail

Reply:
done in 130ba7c692abe1b40ea04d8737ce49b6af82a385